File: java/side exercises/darts/src/main/java/Darts.java
class Darts {
	
	private double x;
	private double y;
	
    Darts(double x, double y) {
        this.x = x;
		this.y = y;
    }

    int score() {
        double radius = getRadius (x, y);
		
		return (radius > 10 ? 0 : (radius > 5 ? 1 : (radius > 1 ? 5 : 10)));
    }
	
	private double getRadius (double x, double y) {
		return Math.sqrt(x * x + y * y);
	}

}

File: csharp/side exercises/darts/Darts.cs
using System;

public static class Darts
{
    private const int OuterCircle = 10;
    private const int MiddleCircle = 5;
    private const int InnerCircle = 1;
    private const int OutsideTaget = 0;
    private const int InOuterCircle = 1;
    private const int InMiddleCircle = 5;
    private const int InInnerCircle = 10;
    
    public static int Score(double x, double y)
    {

        double distance = x * x + y * y;

        if (distance > OuterCircle * OuterCircle)
           return OutsideTaget;

        if (distance > MiddleCircle * MiddleCircle)
           return InOuterCircle;
        
        return distance > InnerCircle ? InMiddleCircle : InInnerCircle;
    }
}
